docs/time_series_causal_graph.md

```diff
@@ -0,0 +1,349 @@
+cg.add_edge('X2 lag(n=1)', 'X2', edge_type=EDGE_T.DIRECTED_EDGE)
+cg.add_edge('X1', 'X3', edge_type=EDGE_T.DIRECTED_EDGE)
+
+ts_cg = TimeSeriesCausalGraph.from_causal_graph(cg)
+```
+
+The time series causal graph will have the same nodes and edges as the causal graph,
+but will be aware of the time information so 'X1 lag(n=1)' and 'X1' represent the same
+variable but at different times.
+
+### From an adjacency matrix
+You can initialize a `cai_causal_graph.time_series_causal_graph.TimeSeriesCausalGraph` instance from an adjacency matrix
+and optionally a list of node names.
+
+```python
+import numpy
+from cai_causal_graph import TimeSeriesCausalGraph
+
+# The adjacency matrix should be a squared binary numpy array
+adjacency_matrix: numpy.ndarray
+
+# Simply via the adjacency matrix
+ts_cg = TimeSeriesCausalGraph.from_adjacency_matrix(adjacency=adjacency_matrix)
+
+# Also specifying the node names
+ts_cg = TimeSeriesCausalGraph.from_adjacency_matrix(adjacency=adjacency_matrix, node_names=['X1 lag(n=1)', 'X1', 'X2 lag(n=1)', 'X2', 'X3'])
+```
+
+### From multiple adjacency matrices
+You can initialize a `cai_causal_graph.time_series_causal_graph.TimeSeriesCausalGraph` instance
+from a dictionary of adjacency matrices. Keys are the time deltas.
+For example, the adjacency matrix with time delta -1 is stored in adjacency_matrices[-1] as would correspond to X-1 -> X,
+where X is the set of nodes.
+
+Example:
+adjacency_matrices = {
+...     -2: numpy.array([[0, 0, 0], [1, 0, 0], [0, 0, 1]]),
+...     -1: numpy.array([[0, 1, 0], [1, 0, 0], [0, 0, 0]]),
+...     0: numpy.array([[0, 1, 1], [0, 0, 1], [0, 0, 0]]),
+... }
+
+
+```python
+from cai_causal_graph import TimeSeriesCausalGraph
+adjacency_matrices = {
+    -2: numpy.array([[0, 0, 0], [1, 0, 0], [0, 0, 1]]),
+    -1: numpy.array([[0, 1, 0], [1, 0, 0], [0, 0, 0]]),
+    0: numpy.array([[0, 1, 1], [0, 0, 1], [0, 0, 0]]),
+}
+
+# Simply via the adjacency matrices
+ts_cg = TimeSeriesCausalGraph.from_adjacency_matrices(adjacency_matrices=adjacency_matrices)
+
+# Also specifying the variable names
+ts_cg = TimeSeriesCausalGraph.from_adjacency_matrices(adjacency_matrices=adjacency_matrices, variable_names=['X', 'Y', 'Z'])
+```
+
+Let the nodes of this example be X,Y,Z. The corresponding edges are the following:
+edges = [
+...     (Z-2, Y),
+...     (X-1, X),
+...     (Y-1, X),
+...     (Y-1, Y),
+...     (Z-1, Z),
+...     (X, Y),
+...     (X, Z),
+...     (Y, Z),
+... ]
+
+## Minimal graph
+The minimal graph is the graph with the minimal number of edges that is equivalent to the original graph.
+In other words, it is a graph that has no edges whose destination is not time delta 0.
+
+
+```python
+from cai_causal_graph import TimeSeriesCausalGraph
+
+ts_cg: TimeSeriesCausalGraph
+
+# Get the minimal graph
+minimal_graph = ts_cg.get_minimal_graph()
+
+# Check whether a graph is in its minimal form
+is_minimal = ts_cg.is_minimal_graph()
```

Comment:
Should say this this returns true.

Reply:
As we don't know the return value, I added
`Returns True if the graph is a minimal graph, False otherwise.`

Reply:
Ah sorry, I thought you were asking if the `minimal_graph` was minimal which would obviously be `True`. Missed that it was the original graph. Would show that `minimal_graph` is `True` though.